### PR TITLE
Refactor/252 ICAccount info

### LINKED ISSUE #252: Merge ICServerInfo and ICAuthenticationInfo into ICAccountInfo
Maybe even change the usage of endpoint from instance variables to an instance of `ICAccountInfo`

### PR DESCRIPTION
**Pull request by timgarrels**
This is an attempt to merge `ICServerInfo` and `ICAuthenticationInfo`.

This fixed #252

### CODE REVIEW

File: packages/IMAPClient-Core.package/ICAccountInfo.class/class/newWithAccountName.host.port.username.password.ssl..st [hunk 1]
```diff
@@ -0,0 +1,10 @@
+instance creation
+newWithAccountName: name host: host port: port username: username password: password ssl: ssl
```
Comment (Paulpanther) [inline packages/IMAPClient-Core.package/ICAccountInfo.class/class/newWithAccountName.host.port.username.password.ssl..st]: Parameter names not squeakish. Please use `aNameString`...

File: packages/IMAPClient-Core.package/ICAccountInfo.class/instance/isValid.st [hunk 2]
```diff
@@ -0,0 +1,4 @@
+validation
+isValid
+
+	^ (self host ~= self class invalidHost and: [self port ~= self class invalidPort]) and: (self username ~= self class invalidUsername and: [self password ~= self class invalidPassword])
```
Comment (Paulpanther) [inline packages/IMAPClient-Core.package/ICAccountInfo.class/instance/isValid.st]: Hard to read, use indentation

File: packages/IMAPClient-Core.package/ICAccountInfo.class/instance/username.st [hunk 3]
```diff
@@ -0,0 +1,4 @@
+accessing
+username
+
+	^ username ifNil: [self username: self class invalidUsername].
```
Comment (Paulpanther) [inline packages/IMAPClient-Core.package/ICAccountInfo.class/instance/username.st]: Why not `... [username := ...`? The check which happens in `username:` is not needed for the initial value and this might cause problems if the check in `username:` is extended to a validity check like in `isValid`

File: packages/IMAPClient-Core.package/ICAccountInfo.class/instance/password.st [hunk 4]
```diff
@@ -0,0 +1,4 @@
+accessing
+password
+	
+	^ password ifNil: [self password: self class invalidPassword]
```
Comment (Paulpanther) [inline packages/IMAPClient-Core.package/ICAccountInfo.class/instance/password.st]: Why not `... [password := ...`? The check in which happens `password:` is not needed for the initial value and this might cause problems if the check in `password:` is extended to a validity check like in `isValid`

File: packages/IMAPClient-Core.package/ICAccountInfo.class/properties.json [hunk 5]
```diff
@@ -4,11 +4,15 @@
...
 	"instvars" : [
+		"accountName",
+		"host",
+		"port",
 		"username",
-		"password" ],
-	"name" : "ICAuthenticationInfo",
+		"password",
+		"ssl" ],
+	"name" : "ICAccountInfo",
```
Comment (Paulpanther) [inline packages/IMAPClient-Core.package/ICAccountInfo.class/properties.json:15]: Write a class comment pls

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/asDictionary.st [hunk 6]
```diff
@@ -7,11 +7,11 @@ asDictionary
 	
 	dataDictionary := Dictionary new.
 	
-	dataDictionary at: 'ssl' put: (self ssl).
-	dataDictionary at: 'name:' put: (self name).
-	dataDictionary at: 'address' put: (self serverInfo host).
-	dataDictionary at: 'port' put: (self serverInfo port).
-	dataDictionary at: 'username' put: (self authenticationInfo username).
-	dataDictionary at: 'password' put: (self authenticationInfo password).
+	dataDictionary at: 'ssl' put: (self accountInfo ssl).
```
Comment (Paulpanther) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/asDictionary.st]: `ICAccountInfo` should implement `asDictionary` since this is more intuitive

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/connectWith..st [hunk 7]
```diff
@@ -0,0 +1,11 @@
+connection
+connectWith: aStreamClass
+
+	self accountInfo isValid ifFalse: [^ Error signal: self class errorInvalidConnectionConfiguration].
+ 
+	[self stream
+		ssl: self useSSL;
```
Comment (Paulpanther) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/connectWith..st]: Why not `self accountInfo ssl`?

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/fromAccountInfo..st [hunk 8]
```diff
@@ -0,0 +1,14 @@
+accessing
+fromAccountInfo: anICAccountInfo
```
Comment (Paulpanther) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/fromAccountInfo..st]: This is a very bad name. Please change this to `connectAndLoginWithAccountInfo:` or something similar. And why is this accessing?

File: packages/IMAPClient-Core.package/ICEndPoint.class/instance/useSSL.st [hunk 9]
```diff
@@ -1,4 +1,4 @@
 accessing
 useSSL
```
Comment (Paulpanther) [inline packages/IMAPClient-Core.package/ICEndPoint.class/instance/useSSL.st]: Pls delete

**Review by Paulpanther — DISMISSED** [review]
- [ ] `canConvert` should be inclusive (Define all symbols which are allowed) and not exclusive (Define all symbols which aren't allowed), since this might miss some edge cases
- [ ] Please add an test for `canConvert`

... This review is done by @felixrindt and me

**Comment by timgarrels** [comment]
>     * [ ]  `canConvert` should be inclusive (Define all symbols which are allowed) and not exclusive (Define all symbols which aren't allowed), since this might miss some edge cases
> 
>     * [ ]  Please add an test for `canConvert`
> 
> 
> ... This review is done by @felixrindt and me

I created #258 to cover this. I do not think that these request are fit for this PR

**Review by Paulpanther — APPROVED** [review]
